### PR TITLE
UCT/IB/WAKEUP: Split CQ in wakeup_arm

### LINKED ISSUE #1039: UCX hangs while ucp_worker_destroy (real code callstack)
There is client and server sending/responding each other (with tag_send_nb). After huge number of send/recv operations client decides to finalize its work and synchronizes with server: after ucx endpoints are deleted (and before any worker was destroyed), client and server are synchronized by means of TCP.

These are UCX outputs from client and server.
Client:
```
[1475679520.511113] [r-aa-fatty21:8846 :4]      eager_snd.c:85   UCX  DEBUG pack eager_middle paylen 8183
[1475679520.511116] [r-aa-fatty21:8846 :4]      eager_snd.c:85   UCX  DEBUG pack eager_middle paylen 8183
[1475679520.511119] [r-aa-fatty21:8846 :4]      eager_snd.c:85   UCX  DEBUG pack eager_middle paylen 8183
[1475679520.511122] [r-aa-fatty21:8846 :4]      eager_snd.c:85   UCX  DEBUG pack eager_middle paylen 8183
TRACE: 10-05 14:58:40:   * 0 client.cpp:173] EchoClient is going to quit
TRACE: 10-05 14:58:40:   * 0 client.cpp:175] AVG = 935 client_latency_percentiles = "[392,494,656,91003]" Max = 1074628
TRACE: 10-05 14:58:40:   * 0 :133] Dereference Socket 1 with fd 16
TRACE: 10-05 14:58:40:   * 0 :171] Remove consumer 16
TRACE: 10-05 14:58:40:   * 0 :693] closing2 16
TRACE: 10-05 14:58:40:   * 0 base/ucx_transport.cpp:102] Initiate disconnect
TRACE: 10-05 14:58:40:   * 0 base/ucx_transport.cpp:111] probing ...
TRACE: 10-05 14:58:40:   * 0 base/ucx_transport.cpp:119] probing result 0; sender tag = 0
TRACE: 10-05 14:58:41:   * 0 base/ucx_transport.cpp:122] probing result 0; sender tag = 0
TRACE: 10-05 14:58:42:   * 0 base/ucx_transport.cpp:125] probing result 0; sender tag = 0
TRACE: 10-05 14:58:42:   * 0 base/ucx_transport.cpp:132] last send!
TRACE: 10-05 14:58:42:   * 0 base/ucx_transport.cpp:144] last send end!
[1475679522.511631] [r-aa-fatty21:8846 :0]         ucp_ep.c:222  UCX  DEBUG disconnect ep 0x2d76620
[1475679522.511635] [r-aa-fatty21:8846 :0]         ucp_ep.c:213  UCX  DEBUG destroy ep 0x2d76620 lane 0 uct_ep 0x2d767c0
[1475679522.512036] [r-aa-fatty21:8846 :0]      callbackq.c:75   UCX  TRACE cbq 0x2c18c08: remove 0x2c18cb8 (arg=0x2c19be0) [start:0x2c18cb8 end:0x2c18ce8]
[1475679522.512044] [r-aa-fatty21:8846 :0]          rc_ep.c:173  UCX  DEBUG destroy rc ep 0x2d767c0
TRACE: 10-05 14:58:42:   * 0 base/ucx_transport.cpp:147] begin barrier!
TRACE: 10-05 14:58:44:   * 0 base/ucx_transport.cpp:158] Successfully exchanged barriers barrier msg from client


TRACE: 10-05 14:58:44:   * 0 base/ucx_transport.cpp:163] UcxEndPoint destructor begin

TRACE: 10-05 14:58:44:   * 0 base/ucx_transport.cpp:172] UcxEndPoint destructor finish

TRACE: 10-05 14:58:44:   * 0 base/ucx_transport.cpp:272] UcxTransport destructor begin
[1475679524.513574] [r-aa-fatty21:8846 :0]     ucp_worker.c:80   UCX  DEBUG worker 0x2c17160: remove active message handlers
[1475679524.513585] [r-aa-fatty21:8846 :0]     ucp_worker.c:407  UCX  DEBUG worker 0x2c17160: destroy all endpoints
[1475679524.513591] [r-aa-fatty21:8846 :0]          mpool.c:122  UCX  DEBUG mpool rc_verbs_short_desc destroyed
[1475679524.514936] [r-aa-fatty21:8846 :0]          mpool.c:122  UCX  DEBUG mpool rc_send_desc destroyed
[1475679524.517369] [r-aa-fatty21:8846 :0]          mpool.c:122  UCX  DEBUG mpool rc_recv_desc destroyed
[1475679524.517374] [r-aa-fatty21:8846 :0]       ib_iface.c:480  UCX  WARN  ibv_destroy_cq(recv_cq) 1
```

Server:
```
[1475679520.477990] [r-aa-fatty22:15236:2]      eager_snd.c:85   UCX  DEBUG pack eager_middle paylen 8183
[1475679520.477992] [r-aa-fatty22:15236:2]      eager_snd.c:85   UCX  DEBUG pack eager_middle paylen 8183
[1475679520.477995] [r-aa-fatty22:15236:2]      eager_snd.c:85   UCX  DEBUG pack eager_middle paylen 8183
[1475679520.477997] [r-aa-fatty22:15236:2]      eager_snd.c:85   UCX  DEBUG pack eager_middle paylen 8183
TRACE: 10-05 14:58:42:   * 0 :85] ucx length == 0
TRACE: 10-05 14:58:42:   * 0 base/ucx_transport.cpp:49] Accept disconnect
TRACE: 10-05 14:58:42:   * 0 base/ucx_transport.cpp:57] last send end!
TRACE: 10-05 14:58:42:   * 0 base/ucx_transport.cpp:59] probing ...
TRACE: 10-05 14:58:42:   * 0 base/ucx_transport.cpp:70] probing result 0; sender tag = 0
TRACE: 10-05 14:58:43:   * 0 base/ucx_transport.cpp:75] probing result 0; sender tag = 0
TRACE: 10-05 14:58:44:   * 0 base/ucx_transport.cpp:80] probing result 0; sender tag = 0
[1475679524.511967] [r-aa-fatty22:15236:2]         ucp_ep.c:222  UCX  DEBUG disconnect ep 0x2a15180
[1475679524.511979] [r-aa-fatty22:15236:2]         ucp_ep.c:213  UCX  DEBUG destroy ep 0x2a15180 lane 0 uct_ep 0x2a15320
[1475679524.512676] [r-aa-fatty22:15236:2]      callbackq.c:75   UCX  TRACE cbq 0x266ebe8: remove 0x266ec98 (arg=0x266fb40) [start:0x266ec98 end:0x266ecc8]
[1475679524.512700] [r-aa-fatty22:15236:2]          rc_ep.c:173  UCX  DEBUG destroy rc ep 0x2a15320
TRACE: 10-05 14:58:44:   * 0 base/ucx_transport.cpp:84] begin barrier!
TRACE: 10-05 14:58:44:   * 0 base/ucx_transport.cpp:95] Successfully exchanged barriers barrier msg from client


TRACE: 10-05 14:58:44:   * 0 base/ucx_transport.cpp:163] UcxEndPoint destructor begin

TRACE: 10-05 14:58:44:   * 0 base/ucx_transport.cpp:172] UcxEndPoint destructor finish
```

Hang process stacktrace:

```
#0  0x00007f7fd84b56d5 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib64/libpthread.so.0
#1  0x00007f7fd6070c3b in ibv_cmd_destroy_cq () from /lib64/libibverbs.so.1
#2  0x00007f7fd003a399 in mlx4_destroy_cq () from /lib64/libmlx4-rdmav2.so
#3  0x00007f7fd607751e in ibv_destroy_cq () from /lib64/libibverbs.so.1
#4  0x00007f7fd8035666 in uct_ib_iface_t_cleanup (self=0x2d63be0) at ib/base/ib_iface.c:481
#5  0x00007f7fd7b24bc6 in ucs_class_call_cleanup_chain (cls=cls@entry=0x7f7fd8280d00 <uct_rc_verbs_iface_t_class>, obj=obj@entry=0x2d63be0, limit=limit@entry=-1) at type/class.c:50
#6  0x00007f7fd803f458 in uct_rc_verbs_iface_t_delete (self=0x2d63be0) at ib/rc/verbs/rc_verbs_iface.c:177
#7  0x00007f7fd8290140 in ucp_worker_close_ifaces (worker=<optimized out>) at core/ucp_worker.c:29
#8  ucp_worker_destroy (worker=0x2d61160) at core/ucp_worker.c:416
#9  0x000000000064975c in base::UcxTransport::~UcxTransport (this=0xcc4220 <c::ucp_worker>, __in_chrg=<optimized out>) at base/ucx_transport.cpp:238
#10 0x00007f7fd62bae69 in __run_exit_handlers () from /lib64/libc.so.6
#11 0x00007f7fd62baeb5 in exit () from /lib64/libc.so.6
#12 0x00007f7fd62a3b1c in __libc_start_main () from /lib64/libc.so.6
#13 0x00000000004176b1 in _start ()
```

### PR DESCRIPTION
**Pull request by shssf**
This PR fixes #1039

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2553/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1904/ for details (Mellanox internal link).

File: src/uct/ib/base/ib_iface.c [hunk 1]
```diff
@@ -689,23 +689,34 @@ ucs_status_t uct_ib_iface_query(uct_ib_iface_t *iface, size_t xport_hdr_len,
...
+                ++send_cq_count;
+            }
+            if (iface->recv_cq == cq) {
+                ++recv_cq_count;
+            }
+        }
     } while (res == 0);
 
-    if (errno != EAGAIN) {
+    if ((-1 == res) && (errno != EAGAIN)) {
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.c]: why check -1==res? the loop exits only when res!=0 anyway

**Comment by vkocheganov** [comment]
Could you please merge this changes before made test?

**Comment by shssf** [comment]
@yosefe I would propose to merge these changes without test creation (because urgent customer request). I will add tests for this in separate PR.

**Comment by shamisp** [comment]
@shssf - I think it is fine. Just make sure to open issue, so we don't forget.

**Comment by shamisp** [comment]
Before we merge, let's finish this regression run.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2558/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1910/ for details (Mellanox internal link).

**Pull request merged**